core/src/main/scala/za/co/absa/spline/agent/AgentBOM.scala

```diff
@@ -63,15 +61,36 @@ object AgentBOM {
     }
 
     override lazy val postProcessingFilter: Option[PostProcessingFilter] = {
-      tryObtainAll[PostProcessingFilter](ConfProperty.RootPostProcessingFilter)
-        .map(new CompositePostProcessingFilter(_))
-        .toOption
+      val nonDefaultRefs = configs.flatMap(_.getOptionalObject[AnyRef](ConfProperty.RootPostProcessingFilter))
+      val refs =
+        if(nonDefaultRefs.nonEmpty) {
+          nonDefaultRefs
+        } else {
+          defaultConfig.getOptionalObject[AnyRef](ConfProperty.RootPostProcessingFilter)
+            .map(Seq(_))
+            .getOrElse(Seq.empty)
+        }
+
+      val filters = refs.map(obtain[PostProcessingFilter](ConfProperty.RootPostProcessingFilter, _))
+
+      if (filters.nonEmpty) {
+        Some(new CompositePostProcessingFilter(filters))
+      } else {
+        None
+      }
```

Comment:
```suggestion
      filters.asOption.map({
        case Seq(filter) => filter
        case _ => new CompositePostProcessingFilter(filters)
      })
```

The method `TraversableWrapper.asOption` replaces the `if  (xs.nonEmpty)`. Also we probably do not need to wrap a single filter with a composite.